test/integration/components/sqlclient/gosqlclient.go

```diff
@@ -0,0 +1,58 @@
+package main
+
+import (
+	"database/sql"
+	"fmt"
+	"time"
+
+	_ "github.com/lib/pq"
+)
+
+const (
+	host     = "localhost"
+	port     = 5432
+	user     = "postgres"
+	password = "mysecretpassword"
+	dbname   = "testdb"
+)
+
+func main() {
+	psqlconn := fmt.Sprintf("host=%s port=%d user=%s password=%s dbname=%s sslmode=disable", host, port, user, password, dbname)
+
+	db, err := sql.Open("postgres", psqlconn)
+	CheckError(err)
+	defer db.Close()
+
+	_, e := db.Exec("DROP TABLE IF EXISTS students")
+	CheckError(e)
+
+	_, e = db.Exec("CREATE TABLE students (name	VARCHAR(80), id INT)")
+	CheckError(e)
+
+	insertStudent := `INSERT INTO students (name, id) VALUES ($1, $2)`
+	_, e = db.Exec(insertStudent, "Bob", 1)
+	CheckError(e)
+	_, e = db.Exec(insertStudent, "Alice", 2)
+	CheckError(e)
+
+	// Do queries for >5 seconds (it may take Beyla that long to find this process)
+	for i := 1; i < 6; i++ {
+		rows, e := db.Query("SELECT * FROM students")
+		CheckError(e)
+		defer rows.Close()
```

Comment:
I wonder, as a non-DB guy: what is a more representative trace?
- Since the `Query` function starts until it ends.
- Since the `Query` function starts until we close the `rows` object.

Reply:
Excellent points (from further above comment) about prepared statements.  Good food for thought for future improvements, to at least make spans for the DB transactions even if we can't see retrieve the query string.
I also see that queryDC takes additional arguments for placeholder values.  Since we only look at the query string, we might only find "?" placeholders.
I'm not sure what's a better representative trace, but it seems like the query.
Lots of room for future improvement, thanks for the ideas!